test/unit/cache/sfCacheDriverTests.class.php

```diff
@@ -18,7 +18,8 @@ static public function launch($t, $cache)
     $t->is($cache->get('test'), $data, '->get() retrieves data form the cache');
     $t->is($cache->has('test'), true, '->has() returns true if the cache exists');
 
-    $t->ok($cache->set('test', $data, -10), '->set() takes a lifetime as its third argument');
```

Comment:
Not sure if we need to keep the support for negative lifetime.

Also, 0 means for APC forever.

What do your think?

Reply:
At least from a sfAPCCache viewpoint, I don't see a particular use for negative TTLs (which from the APC side, is not a documented behavior as a way to invalidate a key).

Looking through the [docs](https://symfony.com/legacy), I didn't find any reference about using negative TTL  that could lead developers to trust that behavior.

Hence, I think negative TTLs were used here only for convenience of testing. This particular test is to make sure keys expire — not that negative TTLs have that effect. One could actually argue that the use of sleep() is actually a better way to test that keys do expire over time using a regular (eg: positive) TTL. Negative TTLs sure are convenient for testing, and allow faster test runs, but positive TTLs and sleep() actually makes sure keys expire with the most common use (positive TTLs).

In any case, mind we are not changing sfAPCCache behavior here, so there's no BC break here. The only change is the way we test key expiration.